app/models/article.rb

```diff
@@ -30,17 +32,18 @@ class Article < ActiveRecord::Base
   has_many :articles_courses, class_name: ArticlesCourses
   has_many :courses, -> { uniq }, through: :articles_courses
   has_many :assignments
+  belongs_to :wiki
+
+  # Allows us to directly #update from the Tools response
+  alias_attribute :page_id, :native_id
 
   scope :live, -> { where(deleted: false) }
   scope :current, -> { joins(:courses).merge(Course.current).uniq }
   scope :namespace, -> ns { where(namespace: ns) }
 
-  # Always save titles with underscores instead of spaces, since that's the way
-  # they are in the MediaWiki database.
   validates :title, presence: true
-  before_validation do
-    self.title = title.tr(' ', '_')
-  end
+
+  before_validation :set_defaults_and_normalize
```

Comment:
The data needs to be in the database proper, cos we join on it, select values from the column, and so on.  As for `before_validate` vs `after_initialize` (in your comment below), I agree that after_initialize would make more sense, but I believe there were two issues.  One is that, in theory I'd like the normalized + default values to be available to any other hooks, so setting early in the lifecycle is better.  The second issue is probably just my own confusion, I seem to remember tests failing like crazy if I used the after_initialize hook.  Which in hindsight doesn't sound right, maybe I hadn't figured out the build (no hooks) vs create (runs hooks) distinction at that point.

I'll review whether I can move defaults to the after_initialize hook.